XRTK.Anchors/Packages/com.xrtk.anchors/Runtime/Interfaces/IMixedRealityAnchorDataProvider.cs

```diff
@@ -0,0 +1,183 @@
+
+using System;
+using UnityEngine;
+using XRTK.Definitions.Anchors;
+
+namespace XRTK.Interfaces.Anchors
+{
+    /// <summary>
+    /// Interface contract for specific identity provider implementations for use in the <see cref="IMixedRealityAnchorSystem"/>.
+    /// </summary>
+    public interface IMixedRealityAnchorDataProvider : IMixedRealityExtensionDataProvider
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        SupportedAnchorType AnchorType { get; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        bool IsRunning { get; }
+
+        #region Public Methods
+
+        /// <summary>
+        /// 
+        /// </summary>
+        void StartAnchorProvider();
+
+        /// <summary>
+        /// 
+        /// </summary>
+        void StopAnchorProvider();
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="objectToAnchor"></param>
+        /// <param name="position"></param>
+        /// <param name="rotation"></param>
+        /// <param name="timeToLive"></param>
+        void CreateAnchor(GameObject objectToAnchor, Vector3 position, Quaternion rotation, System.DateTimeOffset timeToLive);
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        bool FindAnchor(string id);
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="ids"></param>
+        /// <returns></returns>
+        bool FindAnchors(string[] ids);
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="searchType"></param>
+        /// <returns></returns>
+        bool FindAnchors(AnchorSearchType searchType);
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="objectToAnchorPrefab"></param>
+        /// <returns></returns>
+        bool PlaceAnchor(string id, GameObject objectToAnchorPrefab);
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="anchoredObject"></param>
+        /// <returns></returns>
+        bool HasAnchor(ref GameObject anchoredObject);
+
+        /// <summary>
+        /// Move Native anchor to a new position
+        /// </summary>
+        /// <remarks>
+        /// Moving an object with an existing cloud anchor without referencing it's cached ID destroys the cloud position for the object
+        /// </remarks>
+        /// <param name="anchoredObject">Existing GameObject reference of an object to move</param>
+        /// <param name="worldPos">New world position to move to</param>
+        /// <param name="worldRot">New rotation to apply</param>
+        /// <param name="cloudAnchorID">Preexisting cloud anchorID</param>
+        /// <returns></returns>
+        bool MoveAnchor(ref GameObject anchoredObject, Vector3 position, Quaternion rotation, string cloudAnchorID = "");
```

Comment:
why do we need to `ref` the game object?

Reply:
Recommended in the ASA implementation, to ensure the source GameObject is not altered.  "Just in case". 
Unless you feel it will work without?

Reply:
I usually do the opposite of what MSFT recommends lol. They don't usually do it the proper way.

Reply:
I don't understand what could be altered bypassing the object by reference anyway since Unity has a managed and native version of it

Reply:
From testing, seems to work well, and is a good indicator that the source GameObject is being altered.

Reply:
I just typically don't agree with passing objects by reference in a public api